src/modules/companies/lib/invoice-accounts.js

```diff
@@ -0,0 +1,146 @@
+const waterServiceRepos = require('../../../lib/connectors/repos');
+const crmConnectors = require('../../../lib/connectors/crm-v2');
+
+const invoiceAccountsValidators = require('../validators/invoice-accounts');
+const moment = require('moment');
+const { omit, startCase } = require('lodash');
+
+const getAgentCompany = async agent => {
+  if (agent.companyId) return agent;
+  const data = {
```

Comment:
I think we should probably be hydrating service models with data and then mapping/saving those back to the CRM.  This does involve lots of mapping, but hopefully handles some of the validation/business logic since those models already contain validators.

I think the InvoiceAccount model probably contains the full structure you need so I would aim to hydrate this and then persist it.